backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/query/impl/SourceHitExtractor.java

```diff
@@ -22,7 +22,7 @@
  * A hit extractor that will extract a value from the "_source",
  * and insert it into the projection.
  */
-class SourceHitExtractor<F> implements HitExtractor<ProjectionHitCollector> {
+public class SourceHitExtractor<F> implements HitExtractor<ProjectionHitCollector> {
```

Comment:
I would expect the hit extractors to move to the `projection` package. Or even to be merged with the concept of `SearchProjection`, unless there's a good reason to keep them separate.

Note that it would make a lot of these "public" visibility changes disappear.

Otherwise we're introducing a cyclic dependency between the query and projection package, which is never a good sign.